app/javascript/components/Posts.jsx

```diff
@@ -1,33 +1,89 @@
+  const [reload, setReload] = useState(false); // A hook for activating the useEffect hook to update the index of posts
+  const [toastMessage, setToastMessage] = useState(""); // The message of the toasts that this page displays
+
+  // Submits the current post inside the shown text area to the index
+  //  - Doesn't post if the post has no characters in it or too many characters..\
+  //  - Submits the post to the db index if text is valid.
+  const submitPost = () => {
+    const token = document.querySelector('meta[name="csrf-token"]').content;
+    const toast = new bootstrap.Toast(document.getElementById('postToast'));
+
+    if (currentPost.length === 0) {
+      setToastMessage("Nothing to tweet!")
+      toast.show()
+      return
+    }
+
+    if (currentPost.length > 500) {
+      setToastMessage("Tweet must be under 500 characters!")
+      toast.show()
+      return
+    }
+
+    const url = "/api/v1/posts/create";
+    fetch(url, { method: 'POST', headers: { "X-CSRF-Token": token, 'Content-Type': 'application/json' }, body: JSON.stringify({ tweet: currentPost }) })
+
+    document.getElementById('post').value = "";
+    setCurrentPost("");
+
+    setToastMessage("Tweeted!")
+    toast.show()
+    
+    setReload(!reload) // Trigger useEffect to update all posts
+  }
 
   // Fetchs the posts
   useEffect(() => {
     const url = "/api/v1/posts/index.json";
     fetch(url)
-    .then(response => {
-      if (response.ok) {
-        return response.json();
-      } else {
-        throw new Error("Network response was not ok.");
-      }})
       .then(response => {
-        setPosts(response);
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error("Network response was not ok.");
+        }
       })
-    }, [])
+      .then(response => {
+        setAllPosts(response);
+      })
+  }, [reload]);
 
-    return (
-      <div>
-        <h1>Posts</h1>
-        <ul>{posts.map((post) => <li key={post.id}><Post post={post}/></li>)}</ul>
-        <Link to="create-post">create post</Link>
+  return (
+    <div className="container">
+      {/* Heading */}
+      <div className="row">
+        <h1 className="display-1">Posts</h1>
+      </div>
+      {/* Create post */}
+      <div className="row">
+        <div className="col d-grid">
+          <div className="position-relative">
+            <textarea className="form-control" placeholder="What's happening?" id="post" style={{ resize: "none", height: "140px" }} required onChange={(e) => setCurrentPost(e.target.value)} />
+            <span className={`position-absolute badge rounded-pill ${(500 - currentPost.length < 0) ? "bg-danger" : "bg-secondary"}`} style={{ bottom: "8px", right: "16px" }}>{500 - currentPost.length}</span>
+          </div>
+          <button className="btn btn-primary" onClick={submitPost} >Tweet</button>
+        </div>
+      </div>
+      {/* List of all posts */}
```

Comment:
The fact that you feel the need to comment what each section is doing highlights a problem in your code. If there are distinct sections that are nameable you should be able to split them in code so that they actually have a name. I'm not sure what the best way to do this is. It's probably creating components for each of them but this could lead to problems with your toast message.

Reply:
Got rid of the comments but don't think I need to separate any specific thing into its own component. I think it's pretty clear what each section does, I just originally commented those at the start to as a placeholder to figure out the layout of the page (^^)b. I've separated the row sections out in the code